lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -299,6 +299,28 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
       return nullptr;
     Value cdf = buildUnitNormalCdf(b, loc, payloadArgs[0]);
     return b.create<arith::MulFOp>(loc, payloadArgs[0], cdf);
+  }
+    if (auto prelu = dyn_cast<AtenPreluOp>(op)) {
+    if (!prelu.getType()
+             .cast<ValueTensorType>()
+             .getDtype()
+             .isa<mlir::FloatType>()) {
+      prelu.emitError("unimplemented: non-floating point dtype");
+      return nullptr;
+    }
+    Type elementType = payloadArgs[0].getType();
+    Value constZero =
+        b.create<arith::ConstantOp>(loc, b.getZeroAttr(elementType));
+    Value pred = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UGT,
+                                         payloadArgs[0], constZero);
+    Value positivePart =
+        b.create<arith::SelectOp>(loc, pred, payloadArgs[0], constZero);
+    Value negativePart =
+        b.create<arith::SelectOp>(loc, pred, constZero, payloadArgs[0]);
+    Value scale = convertScalarToDtype(b, loc, operands[1], elementType);
```

Comment:
This function expects a value with a builtin element type, but `operands[1]` is a tensor. You should be using `payloadArgs[1]` instead. This should've triggered the `assert` here: https://github.com/llvm/torch-mlir/blob/64f9a0e978d530a2c00d2134e6a01c41cb034b9a/lib/Conversion/Utils/Utils.cpp#L291

Make sure you're compiling with debug info so that `asserts` are not removed. Happy to help with any other questions.

Reply:
Thanks, updated the scale value.

Reply:
Hi @ramiro050, Running the e2e test with `python -m e2e_testing.main -f 'ElementwisePreluModule_basic' -v -c linalg` giving segmentation fault - 

Compiling ElementwisePreluModule_basic...
Running ElementwisePreluModule_basic...
[1]    94003 segmentation fault (core dumped)  python -m e2e_testing.main -f 'ElementwisePreluModule_basic' -v -c linalg

how to debug that?

Reply:
With a segfault, the first step is to get a stack trace, to know what part of the code is triggering it, then proceed from there. You can get a stack trace by running the test in gdb:

```
gdb --args python -m e2e_testing.main -f 'ElementwisePreluModule_basic' -v -c linalg
```

Then do use the `r` command to run the test, and once it crashes you can use the `bt` command to get a backtrace

Reply:
Hi @ramiro050,

I have seen the backtrace but I am not sure where to look for _mlir__mlir_ciface_forward. Please help.

```
(gdb) bt
#0  0x00007ffff756549a in _mlir__mlir_ciface_forward ()
#1  0x00007ffff72b39dd in ffi_call_unix64 () from /home/mcw/anaconda3/lib/python3.9/lib-dynload/../../libffi.so.7
#2  0x00007ffff72b3067 in ffi_call_int () from /home/mcw/anaconda3/lib/python3.9/lib-dynload/../../libffi.so.7
#3  0x00007ffff72cd0f6 in _call_function_pointer (argtypecount=<optimized out>, argcount=1, resmem=0x7fffffffc2d0, restype=<optimized out>, atypes=<optimized out>, avalues=<optimized out>, 
    pProc=0x7ffff7565490 <_mlir__mlir_ciface_forward>, flags=4353) at /usr/local/src/conda/python-3.9.7/Modules/_ctypes/callproc.c:920
#4  _ctypes_callproc (pProc=<optimized out>, argtuple=<optimized out>, flags=<optimized out>, argtypes=<optimized out>, restype=<optimized out>, checker=<optimized out>)
    at /usr/local/src/conda/python-3.9.7/Modules/_ctypes/callproc.c:1263
#5  0x00007ffff72c973e in PyCFuncPtr_call (self=<optimized out>, inargs=<optimized out>, kwds=<optimized out>) at /usr/local/src/conda/python-3.9.7/Modules/_ctypes/_ctypes.c:4201
#6  0x00005555556989ef in _PyObject_MakeTpCall (tstate=0x555555915b10, callable=0x7fff9dc8b100, args=<optimized out>, nargs=<optimized out>, keywords=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:191
#7  0x000055555571e8b4 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fff9dc7d960, callable=<optimized out>, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:116
#8  PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#9  call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#10 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7fff9dc7d7c0, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#11 0x00005555556d68e2 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#12 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fffffffc8a8, kwcount=0, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7fff9f4abda0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#13 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#14 0x000055555570295b in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=3, args=0x7fffffffc890, callable=0x7fff9f4aaf70, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#15 method_vectorcall (method=<optimized out>, args=0x7fff9dd291d8, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/classobject.c:83
#16 0x000055555568c8f8 in PyVectorcall_Call (kwargs=<optimized out>, tuple=0x7fff9dd291c0, callable=0x7fff9dc77d40) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:231
#17 _PyObject_Call (tstate=<optimized out>, callable=0x7fff9dc77d40, args=0x7fff9dd291c0, kwargs=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:266
#18 0x0000555555720740 in PyObject_Call (kwargs=0x0, args=0x7fff9dd291c0, callable=0x7fff9dc77d40) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:293
#19 do_call_core (kwdict=0x0, callargs=0x7fff9dd291c0, func=0x7fff9dc77d40, tstate=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5123
#20 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7fff9dc83640, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3580
#21 0x00005555556d6b0d in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#22 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fff9dd1c060, kwcount=0, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x7fff9dd29140, name=<optimized out>, qualname=0x7fff9f4a3810) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#23 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#24 0x000055555568c8f8 in PyVectorcall_Call (kwargs=<optimized out>, tuple=0x7fff9dd1c040, callable=0x7fff9dc7b1f0) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:231
#25 _PyObject_Call (tstate=<optimized out>, callable=0x7fff9dc7b1f0, args=0x7fff9dd1c040, kwargs=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:266
#26 0x0000555555720740 in PyObject_Call (kwargs=0x0, args=0x7fff9dd1c040, callable=0x7fff9dc7b1f0) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:293
#27 do_call_core (kwdict=0x0, callargs=0x7fff9dd1c040, func=0x7fff9dc7b1f0, tstate=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5123
#28 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x55555a4ac820, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3580
#29 0x00005555556d7753 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#30 function_code_fastcall (globals=<optimized out>, nargs=<optimized out>, args=<optimized out>, co=<optimized out>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:330
#31 _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:367
#32 0x000055555564d10d in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x55555a297c38, callable=0x7fffa01879d0, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#33 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#34 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#35 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x55555a297a90, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3504
#36 0x00005555556d68e2 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#37 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fff9dcc7768, kwcount=0, kwstep=1, 
    defs=0x7ffff7413f58, defcount=1, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7ffff7421850) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#38 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#39 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fff9dcc7750, callable=0x7fffaf5f54c0, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#40 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#41 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#42 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7fff9dcc75b0, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#43 0x00005555556d6b0d in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#44 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x55555a297a50, kwcount=0, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x7fff9dd29740, name=<optimized out>, qualname=0x7ffff74218f0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#45 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#46 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x55555a297a48, callable=0x7fff9dcc5940, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
--Type <RET> for more, q to quit, c to continue without paging--
#47 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#48 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#49 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x55555a297860, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#50 0x00005555556d6b0d in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#51 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fff9df3c628, kwcount=0, kwstep=1, 
    defs=0x7ffff7414358, defcount=2, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7ffff74577b0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#52 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#53 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fff9df3c608, callable=0x7fffaf5f5550, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#54 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#55 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#56 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7fff9df3c440, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#57 0x00005555556d6b0d in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#58 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7ffff7591e30, kwcount=0, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7ffff76075f0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#59 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#60 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7ffff7591e30, callable=0x7fff9dcc5670, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#61 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#62 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#63 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7ffff7591cc0, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#64 0x00005555556d68e2 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#65 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x0, kwcount=0, kwstep=2, defs=0x0, defcount=0, 
    kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#66 0x0000555555788bac in _PyEval_EvalCodeWithName (qualname=0x0, name=0x0, closure=0x0, kwdefs=0x0, defcount=0, defs=0x0, kwstep=2, kwcount=0, kwargs=<optimized out>, kwnames=<optimized out>, argcount=<optimized out>, 
    args=<optimized out>, locals=<optimized out>, globals=<optimized out>, _co=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4359
#67 PyEval_EvalCodeEx (_co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kws=<optimized out>, kwcount=0, defs=0x0, defcount=0, kwdefs=0x0, closure=0x0)
    at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4375
#68 0x00005555556d79eb in PyEval_EvalCode (co=<optimized out>, globals=<optimized out>, locals=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:826
#69 0x00005555557c20be in builtin_exec_impl.isra.17 (locals=0x7ffff7600100, globals=0x7ffff7600100, source=0x7ffff7456df0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/bltinmodule.c:1026
#70 builtin_exec (module=<optimized out>, args=0x5555559d49f0, nargs=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/clinic/bltinmodule.c.h:396
#71 0x00005555556aada1 in cfunction_vectorcall_FASTCALL (func=0x7ffff7659d60, args=0x5555559d49f0, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/methodobject.c:430
#72 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555559d49f0, callable=0x7ffff7659d60, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#73 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#74 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#75 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x5555559d4830, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#76 0x00005555556d68e2 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#77 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x5555559adf00, kwcount=0, kwstep=1, 
    defs=0x7ffff7531fa8, defcount=5, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7ffff7536cb0) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#78 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#79 0x000055555564eebb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555559aded8, callable=0x7ffff752d670, tstate=<optimized out>)
    at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:118
#80 PyObject_Vectorcall () at /tmp/build/80754af9/python-split_1631797238431/work/Include/cpython/abstract.h:127
#81 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x555555915b10) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:5075
#82 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x5555559add30, throwflag=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:3518
#83 0x00005555556d68e2 in _PyEval_EvalFrame () at /tmp/build/80754af9/python-split_1631797238431/work/Include/internal/pycore_ceval.h:40
#84 _PyEval_EvalCode (tstate=<optimized out>, _co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7ffff7536728, kwcount=0, kwstep=1, 
    defs=0x7ffff752f1a8, defcount=1, kwdefs=0x0, closure=0x0, name=<optimized out>, qualname=0x7ffff7531d50) at /tmp/build/80754af9/python-split_1631797238431/work/Python/ceval.c:4327
#85 0x00005555556d7527 in _PyFunction_Vectorcall (func=<optimized out>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:396
#86 0x000055555568c8f8 in PyVectorcall_Call (kwargs=<optimized out>, tuple=0x7ffff7536700, callable=0x7ffff74518b0) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:231
#87 _PyObject_Call (tstate=<optimized out>, callable=0x7ffff74518b0, args=0x7ffff7536700, kwargs=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Objects/call.c:266
#88 0x00005555557bd53a in pymain_run_module (modname=<optimized out>, set_argv0=1) at /tmp/build/80754af9/python-split_1631797238431/work/Modules/main.c:297
#89 0x00005555557c1a18 in pymain_run_python (exitcode=0x7fffffffe080) at /tmp/build/80754af9/python-split_1631797238431/work/Modules/main.c:598
#90 Py_RunMain () at /tmp/build/80754af9/python-split_1631797238431/work/Modules/main.c:683
#91 0x00005555557c1d79 in Py_BytesMain (argc=<optimized out>, argv=<optimized out>) at /tmp/build/80754af9/python-split_1631797238431/work/Modules/main.c:1129
#92 0x00007ffff7c5f083 in __libc_start_main (main=0x555555669d80 <main>, argc=8, argv=0x7fffffffe288, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffe278) at ../csu/libc-start.c:308
#93 0x0000555555746bc3 in _start () 



```

Reply:
This is likely due to a runtime assertion in the generated MLIR being triggered

Reply:
Actually, I think the issue is the decorator annotation in the e2e test. See my comment below